### PR TITLE
Tokenize scipy sparse arrays and np.matrix

### PR DESCRIPTION
**Pull request by rth**
Fixes https://github.com/dask/dask/issues/3054

This PR adds tokenization for scipy sparse arrays. Because a sum of such array along one of the dimensions produces a `np.matrix`, this PR also adds tokenization for the latter class.

cc @jcrist

### CODE REVIEW

File: dask/tests/test_base.py [hunk 1]
```diff
@@ -283,6 +284,52 @@ def test_tokenize_base_types(x):
     assert tokenize(x) == tokenize(x), x
 
 
+@pytest.mark.parametrize('cls_name',
+                         ('ndarray', 'matrix',
```
Comment (jcrist) [inline dask/tests/test_base.py]: The implementation looks good to me, but this test is more complicated than it needs to be. We already have tests for `np.ndarray`, would be cleaner if you split out the testing for `np.matrix`, and had this test just focus on `scipy.sparse` objects.

Comment (rth) [reply dask/tests/test_base.py]: You are right, thanks for the review. This should be fixed now.

**Comment by rth** [comment]
The travis build `UPSTREAM_DEV=1` fails with the latest commit in `test_parquet.py`. I suppose that's not due to this PR?

File: dask/base.py [hunk 2]
```diff
@@ -633,7 +633,6 @@ def normalize_dataframe(df):
 def register_numpy():
     import numpy as np
 
-    @normalize_token.register(np.ndarray)
```
Comment (jcrist) [inline dask/base.py]: Please re-add this line, and remove the added line at 664. This change is unnecessary.

Comment (rth) [reply dask/base.py]: I wasn't sure if the decorators had some side effects; as I'm using `normalize_array` inside `normalize_matrix`..  Reverted it following your comment.  Thanks for the review!

**Comment by jcrist** [comment]
Thanks @rth, this is in.

**Pull request merged**